### ISSUE TITLE
Integration issue Java Kubernetes Client

### ISSUE THREAD

**Issue by cesartl** [datawire/telepresence#563]
I'm having issue running Java project which uses the [Kubernetes client](https://github.com/fabric8io/kubeflix).

It seems that when running locally with Telepresence, the client is not able to access the same resources (here I'm guessing client certificates for the K8s master API).

I've created a [sample project](https://github.com/cesartl/telepresence-k8s/tree/master) that demonstrate the issue.

PS: I'm observing something similar using the [AWS Java SDK](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/welcome.html). The SDK is normally "smart enough" to detect when it is running within an EC2 instance. However when my App is running locally within Telepresence it fails to detect the region and credentials and they must be passed manually

**Comment by ark3** [comment]
Thank you for the detailed and easily-reproduced issue. I believe all the trouble is because of this:

```
15:45:01.303 [main] DEBUG io.fabric8.kubernetes.client.Config - Trying to configure client from service account...      
15:45:01.304 [main] DEBUG io.fabric8.kubernetes.client.Config - Did not find service account ca cert at: [/var/run/secrets/kubernetes.io/serviceaccount/ca.crt].                                                                                
15:45:01.305 [main] WARN io.fabric8.kubernetes.client.Config - Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.                                                                       
15:45:01.305 [main] DEBUG io.fabric8.kubernetes.client.Config - Trying to configure client namespace from Kubernetes service account namespace path...                                                                                          
15:45:01.306 [main] DEBUG io.fabric8.kubernetes.client.Config - Did not find service account namespace at: [/var/run/secrets/kubernetes.io/serviceaccount/namespace]. Ignoring.                                                                 
```

The service account information is available at `$TELEPRESENCE_ROOT/var/run/secrets/kubernetes.io/serviceaccount`, which is not quite where the client library is expecting it. Apparent options include [configuring the library via environment variable](https://github.com/fabric8io/kubernetes-client#configuring-the-client) or using Docker volume mounts to put the service account information in the expected location, but those options will not work! Because of the design of Telepresence, you do not know the value of $TELEPRESENCE_ROOT until it's too late. We need to fix #454 to resolve this issue.

I would guess that the AWS issue is similar: the library probably looks for AWS credentials in a known spot on the filesystem, but Telepresence does not put them there.

**Comment by ark3** [comment]
With Telepresence 0.85's new `--mount` feature, I was able to get the Kubernetes client to work as expected. One way is to bind mount the service account stuff into the usual place. In your example I modified `telepresence-discovery.sh` to read:

```
telepresence --mount=/tmp/known --new-deployment telepresence-k8s --docker-run -e PROFILES=discovery --rm -v$(pwd):/build -v $HOME/.m2/repository:/m2 -v=/tmp/known/var/run/secrets:/var/run/secrets -p 8080:8080 maven:3.5.3-jdk-8-slim mvn -Dmaven.repo.local=/m2 -f /build spring-boot:run
```

In short, I added the `--mount` option and an additional `-v` for Docker.

**Comment by cesartl** [comment]
Of course, I would love to!

Where would be the best place for me to do that? I don't have a personal blog.

**Comment by richarddli** [comment]
Thanks! If you don't have a blog, I think the best place for now would be adding a new tutorial to the Tutorials section of the Telepresence docs (https://github.com/datawire/telepresence/tree/master/docs/tutorials). It's fairly straightforward -- just add a Markdown file there, and link to it from the SUMMARY.md document which generates the TOC.

**Comment by cesartl** [comment]
@richarddli I've created PR #614 
Let me know if it needs any tweaking

**Comment by ark3** [comment]
With #454 being fixed by #610 and with the further documentation in #614, I shall close this issue. Thanks for your help @cesartl, both with this excellent issue and associated code, as well as your docs PR.
